lib/chore/worker.rb

```diff
@@ -42,6 +42,24 @@ def expires_at
       @started_at + total_timeout
     end
 
+    def duplicate_work?(item)
+      # if we've got a duplicate, remove the message from the queue by not actually running and also not reporting any errors
+      payload = options[:payload_handler].payload(item.decoded_message)
+      
+      # only dedupe on the message id if we're getting work from an sqs consumer (currently the only consumer that can recieve the same message id twice)
+      return false if item.consumer.class != Chore::Queues::SQS::Consumer
```

Comment:
Could we base this off the presence of dedupe servers rather than the consumer class?